main/integration-tests/__test__/api-tests/workspace-types/get-workspace-types.test.js

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Get workspace-types scenarios', () => {
+  let setup;
+  let adminSession;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Get workspace-types', () => {
+    it('should fail to get approved if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      await adminSession2.resources.users.deactivateUser(adminSession2.user);
+
+      await expect(adminSession2.resources.workspaceTypes.getApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail to get not-approved if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      await adminSession2.resources.users.deactivateUser(adminSession2.user);
+
+      await expect(adminSession2.resources.workspaceTypes.getNotApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail to get approved if user is anonymous', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+
+      await expect(anonymousSession.resources.workspaceTypes.getApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should fail to get not-approved if user is anonymous', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+
+      await expect(anonymousSession.resources.workspaceTypes.getNotApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should return approved if user is researcher', async () => {
+      const researcherSession = await setup.createResearcherSession();
+
+      await expect(researcherSession.resources.workspaceTypes.getApproved()).resolves.toEqual(expect.any(Array));
+    });
+
+    it('should return no not-approved if user is researcher', async () => {
```

Comment:
Nit: The description is a bit confusing, should it return, or should not?

Reply:
Thanks, updated this an other descriptions